internal/manifest/statecheck/deployment_state_check.go

```diff
@@ -0,0 +1,62 @@
+package statecheck
+
+import (
+	apiappsv1 "k8s.io/api/apps/v1"
+	apicorev1 "k8s.io/api/core/v1"
+	"k8s.io/kubectl/pkg/util/deployment"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+)
+
+const (
+	NewReplicaSetAvailableReason = "NewReplicaSetAvailable"
+	FoundNewReplicaSetReason     = "FoundNewReplicaSet"
+	NewReplicaSetCreatedReason   = "NewReplicaSetCreated"
+	ReplicaSetUpdatedReason      = "ReplicaSetUpdated"
+)
+
+type DeploymentStateChecker interface {
+	GetState(deploy *apiappsv1.Deployment) (shared.State, error)
+}
+
+func NewDeploymentStateCheck() *DeploymentStateCheck {
+	return &DeploymentStateCheck{}
+}
+
+type DeploymentStateCheck struct{}
+
+func (*DeploymentStateCheck) GetState(
+	deploy *apiappsv1.Deployment,
+) (shared.State, error) {
+	deploymentState := getDeploymentState(deploy)
+	return deploymentState, nil
```

Comment:
For me personally we could move the contents from o`getDeploymentState` up into this function and avoid the additional "layer", but also fine as is.